openwisp_controller/config/api/serializers.py

```diff
@@ -0,0 +1,205 @@
+import ast
+import collections
+import json
+
+from rest_framework import serializers
+from swapper import load_model
+
+from openwisp_users.api.mixins import FilterSerializerByOrgOwned
+from openwisp_utils.api.serializers import ValidatedModelSerializer
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+Config = load_model('config', 'Config')
+TemplateTag = load_model('config', 'TemplateTag')
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class BaseMeta:
+    read_only_fields = ['created', 'modified']
+
+
+class TemplateSerializer(FilterSerializerByOrgOwned, ValidatedModelSerializer):
+    config = serializers.JSONField()
+    tags = serializers.StringRelatedField(many=True, read_only=True)
+    default_values = serializers.JSONField(required=False, default={})
+
+    class Meta(BaseMeta):
+        model = Template
+        fields = [
+            'id',
+            'name',
+            'tags',
+            'organization',
+            'type',
+            'backend',
+            'default',
+            'required',
+            'default_values',
+            'config',
+            'created',
+            'modified',
+        ]
+
+
+class VpnSerializer(FilterSerializerByOrgOwned, ValidatedModelSerializer):
+    config = serializers.JSONField()
+
+    class Meta(BaseMeta):
+        model = Vpn
+        fields = [
+            'id',
+            'name',
+            'host',
+            'organization',
+            'key',
+            'ca',
+            'cert',
+            'backend',
+            'notes',
+            'dh',
+            'config',
+        ]
+
+
+class JsonConfigField(serializers.Field):
+    def to_representation(self, value):
+        return json.loads(json.dumps(value))
```

Comment:
> I have tried just return json.dumps(value) and it works, can you explain why we have to do json.loads()? Doing it unnecessarily just slows down the rendering.

Yes, you will get the data but the representation of it will be both different in the json response, and the data present in the HTML form field. When we simply use `json.dumps(value)`, the data is not represented in the JSON format instead a single string of all the nested json response will be returned. so to preserve the representation to be same in both of them, I implemented in this manner. PS: Try adding `general`, `interfaces`, and more to see the difference.